src/classes/RD2_ScheduleService.cls

```diff
@@ -185,6 +194,167 @@ public without sharing class RD2_ScheduleService {
+    * @param rds List of Recurring Donation records
+    * @param startDate Date (if null, current date will be used)
+    * @param numInstallments Integer (if null, DEFAULT_VISUALIZATION_SCHEDULE_COUNT will be used)
+    * @param schedulesByRd Map Id, RecurringDonationSchedule__c (if null, getVisualizedInstallments will query for all active schedules)
+    * @return Map<Id, List<Installment>> Recurring Donation Ids to List of upcoming installments
+    */
+    public Map<Id, List<Installment>> getVisualizedInstallments(
+            List<npe03__Recurring_Donation__c> rds, Date startDate, Integer numInstallments, Map<Id, List<RecurringDonationSchedule__c>> schedulesByRd) {
+
+        if (numInstallments == null) {
+            numInstallments = DEFAULT_VISUALIZATION_SCHEDULE_COUNT;
+        }
+
+        if (startDate == null) {
+            startDate = currentDate;
+        }
+
+        Map<Id, List<Installment>> installmentsByRd = new Map<Id, List<Installment>>();
+        if (schedulesByRd == null) {
+            schedulesByRd = getExistingSchedules(rds);
+        }
+
+        for (npe03__Recurring_Donation__c rd : rds) {
+
+            if (rd == null || rd.Id == null) {
+                continue;
+            }
+
+            installmentsByRd.put(rd.Id, doVisualization(startDate, numInstallments, schedulesByRd.get(rd.Id)));
+        }
+
+        currentDate = Date.today();
+
+        return installmentsByRd;
+    }
+
+    /***
+    * @description Calculates the dates of the next n donations for a single Recurring Donation
+    * @param startDate Date
+    * @param numInstallments Integer
+    * @param schedules List of active RecurringDonationSchedule__c records
+    * @return List<Installment>
+    */
+    private List<Installment> doVisualization(Date startDate, Integer numInstallments, List<RecurringDonationSchedule__c> schedules) {
+
+        List<Installment> installments = new List<Installment>();
+        if (schedules == null) {
+            return installments;
+        }
+
+        currentDate = startDate;
+        List <RecurringDonationSchedule__c> filteredSchedules;
+
+        while (installments.size() < numInstallments) {
+
+            // We are changing currentDate with each loop, so the active schedules potentially change with each loop
+            filteredSchedules = new List<RecurringDonationSchedule__c>();
+
+            for (RecurringDonationSchedule__c schedule : schedules) {
+                if (schedule.EndDate__c == null || schedule.EndDate__c >= currentDate) {
+                    filteredSchedules.add(schedule);
+                }
+            }
+
+            Installment ins = getNextInstallment(filteredSchedules);
+            if (ins == null) {
+                break;
+            }
+
+            installments.add(ins);
+            currentDate = ins.nextDonationDate.addDays(1);
+        }
+
+        return installments;
+    }
+
+    /***
+    * @description Wrapper class for Recurring Donation schedule data
```

Comment:
The description here might be better as "Wrapper class for Recurring Donation Installment data"